protocols/floodsub/src/protocol.rs

```diff
@@ -32,18 +32,29 @@ use libp2p_identity::PeerId;
 use libp2p_swarm::StreamProtocol;
 use std::{io, iter, pin::Pin};
 
-const MAX_MESSAGE_LEN_BYTES: usize = 2048;
+pub(crate) const DEFAULT_MAX_MESSAGE_LEN_BYTES: usize = 2048;
 
 const PROTOCOL_NAME: StreamProtocol = StreamProtocol::new("/floodsub/1.0.0");
 
 /// Implementation of `ConnectionUpgrade` for the floodsub protocol.
-#[derive(Debug, Clone, Default)]
-pub struct FloodsubProtocol {}
+#[derive(Debug, Clone)]
+pub struct FloodsubProtocol {
+    /// Maximum message length in bytes.
+    pub max_message_len: usize,
+}
 
 impl FloodsubProtocol {
     /// Builds a new `FloodsubProtocol`.
-    pub fn new() -> FloodsubProtocol {
-        FloodsubProtocol {}
+    pub fn new(max_message_len: usize) -> FloodsubProtocol {
```

Comment:
As `max_message_len` isn't a mandatory parameter, it would be cleaner to have it inside a `Config` (even if it would be the only field), and define a `from_config(config: Config)` constructor. This way, we don't break the interface, and can extend the parameters in the future without breaking the interface.

`new()` could simply call `default()` or `with_config(Config::default())` so that the interface isn't broken for existing users.

Reply:
Good point! Since the field is already present in `FloodsubConfig`, how about we just add a `from_config(config: &FloodsubConfig)` to `FloodsubProtocol`?

Reply:
Let me know if 3bc76e1a works for you.

Reply:
`FloodsubConfig` has extra fields that aren't needed by `FloodsubProtocol`. I am not sure what would be more efficient. wdyt @dariusc93 @jxs

Reply:
> `FloodsubConfig` has extra fields that aren't needed by `FloodsubProtocol`. I am not sure what would be more efficient. wdyt @dariusc93 @jxs

We could probably have a function that accepts only the message length instead of passing `FloodsubConfig` or creating a new struct and passing that instead.

Reply:
> We could probably have a function that accepts only the message length instead of passing `FloodsubConfig`

Did this in e7b335daad9b77f2e11c59b49c8af030d4716e4a, not sure if the setter is actually needed given that there is a single field, and we could also pass `FloodsubProtocol { max_message_len: self.config.max_message_len }` directly to `SubstreamProtocol::new`. Will leave it to you to decide.

Reply:
I would be in favor of either using a new `Config` struct, or having a new constructor e.g `with_max_message_len(...)`

Reply:
See my latest commits, does this work for you?

Reply:
Sorry, I missed the latest commits. Looks good 👍🏻